contrib/plugins/qt-creator-project.py

```diff
@@ -0,0 +1,231 @@
+# Plugin to generate QT-creator project Files
+
+import argparse
+import sys
+import re
+import os
+import shutil
+from os.path import expanduser
+from os.path import join
+from bob.errors import ParseError
+from bob.input import PluginState, PluginProperty, RecipeSet
+from collections import OrderedDict
+from bob.cmds.build import LocalBuilder
+
+def getCheckOutDirs(packages, dirs):
+    # first scan all packages recursive for their dependencies and build a list of checkout dirs
+    for p in packages:
+        deps = p.getDirectDepSteps()
+        path = p.getCheckoutStep().getWorkspacePath()
```

Comment:
Use `p.getCheckoutStep().isValid()` here instead of looking for 'invalid' in the path.